tests/telemetry/test_telemetry.py

```diff
@@ -272,16 +279,22 @@ def invoke_py_cli_from_ptf(ptfhost, cmd, callback):
 
 def test_on_change_updates(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost, gnxi_path):
     logger.info("Testing on change update notifications")
-
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    if duthost.is_supervisor_node():
+        pytest.skip(
+            "Skipping test as no Ethernet0 frontpanel port on supervisor")
     skip_201911_and_older(duthost)
     cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE,
                               submode=SUBMODE_ONCHANGE, update_count=2, xpath="NEIGH_STATE_TABLE",
                               target="STATE_DB")
+    nslist = duthost.get_asic_namespace_list()
+    ns = random.choice(nslist)
 
-    bgp_nbrs = list(duthost.get_bgp_neighbors().keys())
+    bgp_nbrs = list(duthost.get_bgp_neighbors_for_asic(ns).keys())
```

Comment:
Would this still work if duthost is not multi-asic? I believe if it's not multi-asic, the `nslist` will have `[None]` could you double check if your custom implementation of `get_bgp_neighbors_for_asic` will be still functional?

Reply:
As above. Changed logic of get_bgp_neighbors() to handle multi-asic scenario. Use get_bgp_neighbors(ns) here.